tests/utils/logs.go

```diff
@@ -0,0 +1,157 @@
+	"k8s.io/klog/v2"
+)
+
+// EventuallyLogOpts is used to configure EventuallyLogContains
+type EventuallyLogOpts struct {
+	// Pods is the client to interact with the pods (Mandatory)
+	Pods client.PodInterface
+
+	// PodName is the name of the pod to get logs from (Mandatory)
+	PodName string
+
+	// Search is the list of strings to search for in the logs (Mandatory).
+	// The check will succeed if any of the strings is found.
+	Search []string
+
+	// Since is the timepoint after which the logs will be fetched and checked (Optional).
+	Since time.Time
+
+	// Timeout is the maximum time to wait for the target string to appear in the logs (Optional).
+	Timeout time.Duration
+
+	// Tick is the polling interval to fetch and check the logs (Optional).
+	Tick time.Duration
+
+	// ErrorMsg is an optional error message to include in the failure message (Optional).
+	ErrorMsg string
+}
+
+// EventuallyLogContainsAny polls a pod's logs until any of the target strings is found.
+func EventuallyLogContainsAny(opts EventuallyLogOpts) error {
+	var targets [][]byte
+	for _, s := range opts.Search {
+		targets = append(targets, []byte(s))
+	}
+
+	return eventuallyLogContainsFunc(opts, func(logs []byte) bool {
+		for _, t := range targets {
+			if bytes.Contains(logs, t) {
+				return true
+			}
+		}
+		return false
+	})
+}
+
+// EventuallyLogContainsAll polls a pod's logs until all of the target strings is found.
+func EventuallyLogContainsAll(opts EventuallyLogOpts) error {
+	// targets is used as a set, we will remove the target strings from it as they are found.
+	// If the set is empty, we have found all the target strings.
+	targets := make(map[string]struct{})
+	for _, s := range opts.Search {
+		targets[s] = struct{}{}
+	}
+
+	return eventuallyLogContainsFunc(opts, func(logs []byte) bool {
+		for t := range targets {
+			if bytes.Contains(logs, []byte(t)) {
+				delete(targets, t)
+			}
+		}
+
+		return len(targets) == 0
+	})
+}
+
+// eventuallyLogContainsFunc polls a pod's logs until the predicate is satisfied.
+func eventuallyLogContainsFunc(opts EventuallyLogOpts, predicate func(newLogs []byte) bool) error {
+	// Mandatory fields
+	if opts.Pods == nil {
+		return errors.New("Pods client is required")
+	}
+	if opts.PodName == "" {
+		return errors.New("PodName is required")
+	}
+	if len(opts.Search) == 0 {
+		return errors.New("Search is required")
+	}
+
+	// Optional fields
+	if opts.Timeout == 0 {
+		opts.Timeout = 270 * time.Second
+	}
+	if opts.Tick == 0 {
+		opts.Tick = 2 * time.Second
+	}
+
+	// fullLogs is used to generate a more informative error message in case of failure.
+	fullLogs := [][]byte{}
+
+	ctx, cancel := context.WithTimeout(context.Background(), opts.Timeout)
+	defer cancel()
+	tk := time.NewTicker(opts.Tick)
+
+	since := opts.Since
+	for {
+		select {
+		case <-ctx.Done():
+			return logNotFoundErr(opts, ctx.Err(), fullLogs)
+		case <-tk.C:
+		}
+
+		logOpt := &core.PodLogOptions{
+			SinceTime: &meta.Time{Time: since},
```

Comment:
I think the issue might be here? In the api (https://pkg.go.dev/k8s.io/api/core/v1#PodLogOptions) there's a field (`Previous`) that allows returning logs from previously failed containers. Since some of these tests read logs from pods that fail maybe we should set this to true. Not 100% sure though.